app/src/full/kotlin/tunnel/DnsProxy.kt

```diff
@@ -59,25 +61,37 @@ internal class DnsProxy(
             w("failed reading DNS message", e)
             return
         }
+        dnsMessage.question.name
```

Comment:
What does it mean?

Reply:
Looks like I overlooked some code fragment from debugging. I'll wait for the full review and remove it after that. :)